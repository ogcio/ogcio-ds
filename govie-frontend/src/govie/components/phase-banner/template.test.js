const { axe, render, getExamples, htmlWithClassName } = require('../../../../lib/jest-helpers')

describe('Phase banner', () => {
  let examples

  beforeAll(async () => {
    examples = await getExamples('phase-banner')
  })

  describe('by default', () => {
    it('passes accessibility tests', async () => {
      const $ = render('phase-banner', examples.default)

      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('allows additional classes to be added to the component', () => {
      const $ = render('phase-banner', examples.classes)

      const $component = $('.govie-phase-banner')
      expect($component.hasClass('extra-class one-more-class')).toBeTruthy()
    })

    it('renders banner text', () => {
      const $ = render('phase-banner', examples.text)
      const phaseBannerText = $('.govie-phase-banner__text').text().trim()

      expect(phaseBannerText).toEqual('This is a new service â€“ your feedback will help us to improve it')
    })

    it('allows body text to be passed whilst escaping HTML entities', () => {
      const $ = render('phase-banner', examples['html as text'])

      const phaseBannerText = $('.govie-phase-banner__text').html().trim()
      expect(phaseBannerText).toEqual('This is a new service - your &lt;a href="#" class="govie-link"&gt;feedback&lt;/a&gt; will help us to improve it.')
    })

    it('allows body HTML to be passed un-escaped', () => {
      const $ = render('phase-banner', examples.default)

      const phaseBannerText = $('.govie-phase-banner__text').html().trim()
      expect(phaseBannerText).toEqual('This is a new service - your <a href="#" class="govie-link">feedback</a> will help us to improve it.')
    })

    it('allows additional attributes to be added to the component', () => {
      const $ = render('phase-banner', examples.attributes)

      const $component = $('.govie-phase-banner')
      expect($component.attr('first-attribute')).toEqual('foo')
      expect($component.attr('second-attribute')).toEqual('bar')
    })
  })

  describe('with dependant components', () => {
    it('renders the tag component text', () => {
      const $ = render('phase-banner', examples.default)

      expect(htmlWithClassName($, '.govie-phase-banner__content__tag')).toMatchSnapshot()
    })

    it('renders the tag component html', () => {
      const $ = render('phase-banner', examples['tag html'])

      expect(htmlWithClassName($, '.govie-phase-banner__content__tag')).toMatchSnapshot()
    })

    it('renders the tag component classes', () => {
      const $ = render('phase-banner', examples['tag classes'])

      expect(htmlWithClassName($, '.govie-phase-banner__content__tag')).toMatchSnapshot()
    })
  })
})
