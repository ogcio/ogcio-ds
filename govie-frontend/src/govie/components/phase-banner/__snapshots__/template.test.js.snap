// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phase banner with dependant components renders the tag component classes 1`] = `
<strong class="govie-tag govie-phase-banner__content__tag govie-tag--grey">
  alpha
</strong>
`;

exports[`Phase banner with dependant components renders the tag component html 1`] = `
<strong class="govie-tag govie-phase-banner__content__tag">
  <em>
    alpha
  </em>
</strong>
`;

exports[`Phase banner with dependant components renders the tag component text 1`] = `
<strong class="govie-tag govie-phase-banner__content__tag">
  alpha
</strong>
`;
